### ISSUE TITLE
Install error

### ISSUE THREAD

**Issue by QSong-github** [grimbough/rhdf5filters#5]
Hi there,

I tried to install 'rhdf5filters' on my macOS mojave (10.14.6), but has the following issues. I'am using the R version 4.0.3. I have gcc (version: 10.2.0) and hdf5 (version: 1.12.0_1)  installed through homebrew. 

My CPU is Intel(R) Core(TM) i9-9900K CPU @ 3.60GHz.

The errors are shown below:

**BiocManager::install('grimbough/rhdf5filters', ref = "ldflags")
 Bioconductor version 3.12 (BiocManager 1.30.10), R 4.0.3 (2020-10-10)
 Installing github package(s) 'grimbough/rhdf5filters'
 Using github PAT from envvar GITHUB_PAT
 Downloading GitHub repo grimbough/rhdf5filters@ldflags
 ✔  checking for file ‘/private/var/folders/nc/v3q_2p4n5y9_z4ymklnz4wf4nlcfx8/T/RtmpMOSbLg/remotes4eff3a97b975/grimbough-rhdf5filters-4bac582/DESCRIPTION’
 ─  preparing ‘rhdf5filters’:
 ✔  checking DESCRIPTION meta-information
 ─  cleaning src
 ─  running ‘cleanup’
 ─  checking for LF line-endings in source and make files and shell scripts
 ─  checking for empty or unneeded directories
 ─  building ‘rhdf5filters_1.1.1-1.tar.gz’
    Warning: invalid uid value replaced by that for user 'nobody'
    Warning: invalid gid value replaced by that for user 'nobody'

 * installing *source* package ‘rhdf5filters’ ...
 ** using staged installation
 configuring the BLOSC filter...
 checking for gcc... clang
 checking whether the C compiler works... yes
 checking for C compiler default output file name... a.out
 checking for suffix of executables...
 checking whether we are cross compiling... no
 checking for suffix of object files... o
 checking whether we are using the GNU C compiler... yes
 checking whether clang accepts -g... yes
 checking for clang option to accept ISO C89... none needed
 checking whether C compiler accepts -msse2... yes
 checking for gcc __builtin_cpu_init function... no
 configure: error: Need GCC to support X86 CPU features tests
 ERROR: configuration failed for package ‘rhdf5filters’
 * removing ‘/usr/local/lib/R/4.0/site-library/rhdf5filters’
 Error: Failed to install 'rhdf5filters' from GitHub:
   (converted from warning) installation of package ‘/var/folders/nc/v3q_2p4n5y9_z4ymklnz4wf4nlcfx8/T//RtmpMOSbLg/file4eff78ee08b9/rhdf5filters_1.1.1-1.tar.gz’ had non-zero exit status**

**Comment by grimbough** [comment]
This looks like the same issue as in #3 

I'll take a look at it.

**Comment by jing-su** [comment]
I am a co-worker of QSong-github. Just to confirm that by pulling down the github source code and installing using `R CMD INSTALL` with `--configure-args="ax_cv_gcc_check_x86_cpu_init=yes ax_cv_gcc_x86_cpu_supports_sse2=no"`, the package can be successfully installed. 

It is different from #3 that the CPU on the Mac is Intel(R) Core(TM) i9-9900K CPU, which might be an x_86 (?) architecture. 

Another issue is that, when installing other R packages depending on rhdf5fileters using BiocManager, it reported that the installed version (1.3) is not the latest and recommended to install a new version (2.0). Just wonder whether the new version (2.0) is available here. 

Thanks! 

Jing

**Comment by grimbough** [comment]
I suspect that the error message is misleading, and the real issue is that you're using clang rather than gcc, but that ends up tripping the same error message is if it was the wrong architecture.  I'll need to sort the logic in the configure step to pass those two settings automatically when this is the case.

@jing-su could you paste the error message regarding the version numbering.  I've never released a version 2.0 anywhere, and the latest versions on Bioconductor are 1.2 (release) and 1.3 (development).  If you can show me the message I'll try to work out where it's coming from.

**Comment by grimbough** [comment]
Closed due to a lack of additional information. Please reopen if this is still a problem.
